sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java

```diff
@@ -83,6 +78,31 @@ public UpdateResult updateConfiguration(ServiceSpec candidateConfig) throws Conf
                     String.format("Unable to serialize new config to JSON for logging: %s", e.getMessage())));
         }
 
+        if (targetConfig == null) {
+            LOGGER.info("Skipping config diff: There is no old config target to diff against");
+        } else if (candidateConfigJson == null) {
+            LOGGER.info("Skipping config diff: New target couldn't be represented as JSON");
+        } else {
+            // Print a diff of this new config vs the prior config:
```

Comment:
Could we factor out this diff printing code into a method outside of `updateConfiguration`?